python/cudf/benchmarks/common/utils.py

```diff
@@ -0,0 +1,267 @@
+        )
+    elif how == "random":
+        return cudf.Series(rstate.randint(0, len_column, len_gather_map))
+
+
+def make_boolean_mask_column(size):
+    rstate = cupy.random.RandomState(seed=0)
+    return cudf.core.column.as_column(rstate.randint(0, 2, size).astype(bool))
+
+
+def accepts_cudf_fixture(
+    cls, *, dtype="int", nulls=None, cols=None, rows=None, name=None
+):
+    """Pass "standard" cudf fixtures to functions without renaming parameters.
+
+    The fixture generation logic in conftest.py provides a plethora of useful
+    fixtures to allow developers to easily select an appropriate cross-section
+    of the space of objects to apply a particular benchmark to. However, the
+    usage of this fixtures is cumbersome because creating them in a principled
+    fashion results in long names and very specific naming schemes. This
+    decorator abstracts that naming logic away from the developer, allowing
+    them to instead focus on defining the fixture semantically by describing
+    its properties.
+
+    Parameters
+    ----------
+    cls : Union[str, Type]
+        The class of object to test. May either be specified as the type
+        itself, or using the name (as a string). If a string, the case is
+        irrelevant as the string will be converted to all lowercase.
+    dtype : Union[str, Iterable[str]], default 'int'
+        The dtype or set of dtypes to use.
+    nulls : Optional[bool], default None
+        Whether to test nullable or non-nullable data. If None, both nullable
+        and non-nullable data are included.
+    cols : Optional[int], None
+        The number of columns. Only valid if cls == 'dataframe'. If None, use
+        all possible numbers of columns. Specifying multiple values is
+        unsupported.
+    rows : Optional[int], None
+        The number of rows. If None, use all possible numbers of rows.
+        Specifying multiple values is unsupported.
+    fixture : str, default None
+        The name of the fixture as used in the decorated test. If None,
+        defaults to `cls.lower()` if cls is a string, otherwise
+        `cls.__name__.lower()`. Use of this name allows the decorated function
+        to masquerade as a pytest receiving a fixture while substituting the
+        real fixture (with a much longer name).
+
+    Raises
+    ------
+    AssertionError
+        If any of the parameters do not correspond to extant fixtures.
+
+    Examples
+    --------
+    # Note: As an internal function, this example is not meant for doctesting.
+
+    @accepts_cudf_fixture("dataframe", dtype="int", nulls=False, name="df")
+    def bench_columns(benchmark, df):
+        benchmark(df.columns)
+    """
+    if inspect.isclass(cls):
+        cls = cls.__name__
+    cls = cls.lower()
+
+    supported_classes = (
+        "column",
+        "series",
+        "index",
+        "dataframe",
+        "indexedframe",
+        "frame_or_index",
+    )
```

Comment:
RangeIndex is not nullable and can't vary in dtype. In theory we could use the decorator for just sizes, I guess, but that seemed unnecessary. What do you think?

Reply:
If I understand it correctly, including rangeindex here would change:
```python
def test_f(benchmark, rangeindex):
...
```
to
```python
@accept_cudf_fixture("rangeindex", size=[...])
def test_f(benchmark, rangeindex):
...
```
I don't think parameterizing the sizes is particularly useful for rangeindex as the default fixtures should suffice most use cases. I agree that we should leave it as a separate fixture union to avoid the complexity.

Reply:
Yes, you understand correctly and we're in agreement that it doesn't seem very useful.